vim-patch:9.1.{1269,1272}

glepnir opened the pull request:
Problem:  compl_shown_match is updated when starting keyword completion
          and does not include fuzzy matching.
Solution: Do not update compl_shown_match when starting keyword
          completion, since it is the one already selected by the
          keyword completion direction. (glepnir)

closes: vim/vim#17033

https://github.com/vim/vim/commit/e4e4d1c381e9d0af55f6111e9bcaf98ad60461fc


vim-patch:9.1.1272: completion: in keyword completion Ctrl_P cannot go back after Ctrl_N

Problem:  completion: in keyword completion Ctrl_P cannot go back after
          Ctrl_N
Solution: in find_compl_when_fuzzy() always return first match of array, after Ctrl_P
          use compl_shown_match->cp_next instead of compl_first_match.
          (glepnir)

closes: vim/vim#17043

https://github.com/vim/vim/commit/3e50a28a03d136c1e0c1f4fabe50d97faaf08c5c

Fix #33261

glepnir committed:
vim-patch:9.1.1269: completion: compl_shown_match is updated when starting keyword completion

Problem:  compl_shown_match is updated when starting keyword completion
          and does not include fuzzy matching.
Solution: Do not update compl_shown_match when starting keyword
          completion, since it is the one already selected by the
          keyword completion direction. (glepnir)

closes: vim/vim#17033

https://github.com/vim/vim/commit/e4e4d1c381e9d0af55f6111e9bcaf98ad60461fc

Co-authored-by: glepnir <glephunter@gmail.com>

zeertzjq commented:
> Fix https://github.com/neovim/neovim/issues/33261

How?

glepnir commented:
emm I can't reproduce the test case has been covered?

zeertzjq commented:
Yes, but neither the test case nor your comment is related to the problem reported to the issue. Neither of them involve pressing `<C-N>` after `<C-P>`.

glepnir commented:
aha i miss that reprroduced this

glepnir commented:
wait anohter patch we can merge togethor

glepnir committed:
vim-patch:9.1.1272: completion: in keyword completion Ctrl_P cannot go back after Ctrl_N

Problem:  completion: in keyword completion Ctrl_P cannot go back after
          Ctrl_N
Solution: in find_compl_when_fuzzy() always return first match of array, after Ctrl_P
          use compl_shown_match->cp_next instead of compl_first_match.
          (glepnir)

closes: vim/vim#17043

https://github.com/vim/vim/commit/3e50a28a03d136c1e0c1f4fabe50d97faaf08c5c

Co-authored-by: glepnir <glephunter@gmail.com>
